### DISCUSSION TITLE
Consultation : Do you use the Remote CSP Allow List?

### DISCUSSION THREAD

**Discussion by GeekInTheNorth** [GeekInTheNorth/Stott.Security.Optimizely#258] [Polls]
## Reason for Consultation:

At some point in the near future, a refactoring of the Stott Security module will be taking place to split functionality to support both Optimizely SAAS CMS and Optimizely PAAS CMS.  As part of this, I am looking at functionality that has been built, but I suspect is not being used.

## Remote CSP Allow List Functionality

The Remote CSP Allow List was added to support agencies with dedicated Data Teams that would inject common tooling through GTM into websites for multiple clients.  This would allow a JSON file to be hosted somewhere with a fixed list of CSP Sources and which directives they could be applied to the CSP in all of their implementations.  This would allow a team that commonly uses Hotjar to not need to log into the CMS and add Hotjar to the CSP for their clients as this would be automatically added upon detection.

This functionality has not been used within the Agencies I have worked with, the impact of removing it is negligible and it would remove one extra function for refactoring as well as making the internal reporting APIs more responsive. To help me make this decision, please review and respond to the Poll below. 

![CSP Settings Page](https://github.com/user-attachments/assets/95699c29-38ca-4c29-b84a-e8a53b61d228)

**Comment by omartini-dev** [comment]
I don’t have visibility into your specific Stott Security deployments or whether any agencies are actively using the Remote CSP Allow List feature.

However, based on typical usage patterns for this kind of functionality, it is usually only adopted when agencies have a centralized, shared tooling model (e.g., GTM-injected scripts across many client sites) and a deliberate workflow for managing CSP rules externally.

If you have not explicitly onboarded agencies to use a remote JSON-based CSP governance model, and you are not observing:

- Requests in logs fetching or polling a remote allow list
- Configurations in CMS instances pointing to a remote CSP source
- Support or implementation requests referencing automatic CSP injection from external sources

then it is very likely that this feature is not currently in active use.

To make a confident decision about removal, you could:

1. Review logs for any remote allow list fetch activity.
2. Scan existing CMS configurations for enabled remote CSP URLs.
3. Temporarily deprecate the feature behind a flag and monitor for regressions.

So, in response to the poll question:
It is unlikely that this functionality is being actively used, but this should be confirmed via configuration and telemetry before removal.

**Reply by GeekInTheNorth** [reply]
Thank you @omartini-dev for that very well thought out response.

This feature was built while I was working with a Digital Agency where there was a dedicated team where the Agency sold that expertise into multiple clients. 

The challenge is that as this is an Add-on for Optimizely CMS. I build the Add-on, a different entity builds the website and installs it.  This has been installed into numerous instances which are not managed by me or the team I work with.  So I have no access to logs from those environments.

I have deliberated with the idea of implementing usage tracking to get an understanding of what features are actively in use.  I think this introduces it's own challenges around privacy management as well as requiring a tracking tool account or API store for this information that would come at a cost and this add-on does not generate revenue.

My gut feel is that at best, this feature has a small minority usage, but it is likely to be unused.
